src/lib/pickles/composition_types/composition_types.mli

```diff
@@ -1378,33 +1506,43 @@ module Step : sig
       }
     [@@deriving sexp, compare, yojson]
 
+    type 'a vec12 :=
+      ( 'a
+      * ( 'a
+        * ( 'a
+          * ('a * ('a * ('a * ('a * ('a * ('a * ('a * ('a * ('a * unit)))))))))
+          ) ) )
+      Hlist.HlistId.t
+
```

Comment:
At this point, looks like a toplevel type alias might be useful

Reply:
+1